libnetdata/health/health.c

```diff
@@ -138,8 +176,13 @@ int health_silencers_json_read_callback(JSON_ENTRY *e)
                 else if (!strcmp(e->data.string,"DISABLE")) silencers->stype = STYPE_DISABLE_ALARMS;
             } else {
                 debug(D_HEALTH, "JSON: Adding %s=%s", e->name, e->data.string);
-                SILENCER *test = health_silencers_addparam(e->callback_data, e->name, e->data.string);
-                (void)test;
+                SILENCER *silencer = health_silencers_addparam(e->callback_data, e->name, e->data.string);
+                if(silencer) {
+                   if (!helth_is_silencer_linked(silencer))
+                       health_silencers_add(silencer);
+                   else
+                       health_free_sielencer_pointers(silencer);
```

Comment:
Why do you free pointers here? The result of the condition is inconsistent in different cases.

Reply:
This makes things worse I think. The parser will have created a silencer and stored in `e->callback_data` (L121-L123). and  added to the linked list. 

So adding a param (L179) may override values in the silencer structure (memory leak), then it will see that it is linked and remove it (but will remain linked)

Reply:
Thank you @stelfrag to explain!